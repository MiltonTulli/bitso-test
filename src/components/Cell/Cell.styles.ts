import { createUseStyles } from 'react-jss';
import theme, { Theme } from '../../theme';

export default createUseStyles<Theme>({
  root: ({ size }) => ({
    width: size,
    height: size,
    borderRadius: theme.spacing(0.8),
    transition: 'ease-out 0.2s',
    '&:hover': {
      filter: 'opacity(0.8)',
      transform: 'scale(0.95)'
    }
  }),
  active: {
    background: theme.palette.island.active
  },
  inactive: {
    background: theme.palette.island.inactive
  }
});
