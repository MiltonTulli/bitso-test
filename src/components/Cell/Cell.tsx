import React from 'react';
import clsx from 'classnames';
import styles from './Cell.styles';

interface CellProps {
  position: string;
  active: boolean;
  handleToggle: (position: string) => void;
  size: number;
}

export const Cell: React.FunctionComponent<CellProps> = (props: CellProps) => {
  const { position, active, handleToggle, size } = props;
  const classes = styles({ size, active });

  return (
    <div
      data-testid="cell"
      onClick={() => handleToggle(position)}
      className={clsx(classes.root, active ? classes.active : classes.inactive)}
    />
  );
};
