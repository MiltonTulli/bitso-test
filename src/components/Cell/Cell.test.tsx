import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Cell } from './Cell';

describe('<Cell>', () => {
  let props;

  beforeEach(() => {
    props = {
      position: '1-2',
      active: false,
      handleToggle: jest.fn(),
      size: 100
    };
  });

  it('Should match snapshot', () => {
    const wrapper = renderer.create(<Cell {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Should display inactive css', () => {
    const wrapper = shallow(<Cell {...props} />);
    const root = wrapper.find('[data-testid="cell"]');
    expect(wrapper).toHaveLength(1);
    expect(root.hasClass(/inactive/)).toBe(true);
  });
  it('Should display active css', () => {
    props.active = true;
    const wrapper = shallow(<Cell {...props} />);
    const root = wrapper.find('[data-testid="cell"]');
    expect(root.hasClass(/active/)).toBe(true);
  });

  it('Should handle cell click', () => {
    const wrapper = shallow(<Cell {...props} />);
    const root = wrapper.find('[data-testid="cell"]');
    root.simulate('click');
    expect(props.handleToggle).toHaveBeenCalledWith(props.position);
  });
});
