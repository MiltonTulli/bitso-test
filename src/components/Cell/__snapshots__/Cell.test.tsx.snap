// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell> Should match snapshot 1`] = `
<div
  className="root-0-2-1 root-d0-0-2-4 inactive-0-2-3"
  data-testid="cell"
  onClick={[Function]}
/>
`;
